src/integrationtests/java/com/aws/greengrass/integrationtests/certificateauthority/CustomCaConfigurationTest.java

```diff
@@ -0,0 +1,199 @@
+
+        KeyPair intermediateKeyPair = CertificateStore.newRSAKeyPair(2048);
+        X509Certificate intermediateCA = CertificateTestHelpers.arrangeCertificate(
+                new X500Name("CN=root"), new X500Name("CN=intermediate"), intermediateKeyPair, rootKeyPair, true);
+
+        // Service Configuration
+        URI privateKeyUri = new URI("file:///private.key");
+        URI certificateUri = new URI("file:///certificate.pem");
+
+        Topics configurationTopics = Topics.of(new Context(), CLIENT_DEVICES_AUTH_SERVICE_NAME, null);
+        configurationTopics.lookup(CONFIGURATION_CONFIG_KEY, CERTIFICATE_AUTHORITY_TOPIC, CA_PRIVATE_KEY_URI)
+                .withValue(privateKeyUri.toString());
+        configurationTopics.lookup(CONFIGURATION_CONFIG_KEY, CERTIFICATE_AUTHORITY_TOPIC, CA_CERTIFICATE_URI)
+                .withValue(certificateUri.toString());
+
+        CDAConfiguration cdaConfiguration = CDAConfiguration.from(configurationTopics);
+        CertificatesConfig certsConfig = new CertificatesConfig(configurationTopics);
+        certificateManager.updateCertificatesConfiguration(certsConfig);
+
+        // Configure Custom CA
+        when(securityServiceMock.getKeyPair(privateKeyUri, certificateUri)).thenReturn(intermediateKeyPair);
+        when(securityServiceMock.getCertificateChain(privateKeyUri, certificateUri))
+                .thenReturn(new X509Certificate[]{intermediateCA, rootCA});
+        certificateManager.configureCustomCA(cdaConfiguration);
+
+        return new X509Certificate[]{intermediateCA, rootCA};
+    }
+
+
+    /**
+     * Simulates the request client components make to create their certificates. Clients pass their certificate
+     * chain pem to get verified.
+     */
+    private X509Certificate[] arrangeClientComponentCertificateChain() throws CertificateGenerationException,
+            ExecutionException, InterruptedException, TimeoutException {
+        // Generate Client Certificates
+        AtomicReference<CertificateUpdateEvent> eventRef = new AtomicReference<>();
+
+        Pair<CompletableFuture<Void>, Consumer<CertificateUpdateEvent>> asyncCall =
+                TestUtils.asyncAssertOnConsumer(eventRef::set, 1);
+        GetCertificateRequestOptions requestOptions = new GetCertificateRequestOptions();
+        requestOptions.setCertificateType(GetCertificateRequestOptions.CertificateType.CLIENT);
+        GetCertificateRequest certificateRequest =
+                new GetCertificateRequest("com.aws.clients.Plugin", requestOptions, asyncCall.getRight());
+
+        certificateManager.subscribeToCertificateUpdates(certificateRequest);
+        asyncCall.getLeft().get(1, TimeUnit.SECONDS);
+
+        // Arrange the certificate chain leaf comes first, root comes last
+        return ArrayUtils.addAll(new X509Certificate[]{eventRef.get().getCertificate()},
+                eventRef.get().getCaCertificates());
+    }
+
+
+    @Test
```

Comment:
These are the new tests to ensure the custom ca is generating client certs and validating them as expected